org-roam.el

```diff
@@ -374,6 +374,26 @@ current buffer is used."
 (defun org-roam--extract-ref ()
   "Extract the ref from current buffer."
   (cdr (assoc "ROAM_KEY" (org-roam--extract-global-props '("ROAM_KEY")))))
+
+(defun org-roam--ref-type (ref)
+  "Determine the type of the REF from the prefix."
+  (let* ((cite-prefix (org-roam--cite-prefix ref))
+         (is-website (seq-some
+                      (lambda (prefix) (s-prefix? prefix ref))
+                      '("http" "https")))
+         (type (cond (cite-prefix "cite")
+                     (is-website "website")
+                     (t "roam"))))
+    type))
```

Comment:
I wonder if now `org-roam--extract-ref` should return the processed "ref" as well as the "type". I can't think of too many reasons we will want the raw `ROAM_KEY` anymore. 

Merging it might reduce the number of `let*` and `when-let*` required when using the ref

Reply:
Yeah, it might be more efficient to instead write a function that takes in the roam key, and returns a cons cell containing `(type processed-ref)`

Reply:
I agree, that'd be useful for checking the type.  That would allow for a convenient predicate:
```el
(defun org-roam--ref-type-p (type)
  "Return t if the ref from current buffer is TYPE."
  (let ((current (car (org-roam--extract-ref))))
    (eq current type )))
```

Reply:
> > Made all the suggested changes and tested, it works.
> 
> Thanks for that.
> 
> I've checked out the PR, cleared the db, tried to rebuilt it, and here's what I got:
> 
> > EmacSQL had an unhandled condition: "table refs has 2 columns but 3 values were supplied"
> 
> Can you identify the problem?

I fixed by removing the .db file